src/mongo/MongoScrapbook.ts

```diff
@@ -204,4 +207,46 @@ class MongoScriptDocumentVisitor extends MongoVisitor<MongoCommand[]> {
 	protected defaultResult(_node: ParseTree): MongoCommand[] {
 		return this.commands;
 	}
+
+	private parseContext(ctx: mongoParser.ArgumentContext | mongoParser.PropertyValueContext): Object {
+		let parsedObject: Object = {};
+		if (!ctx || ctx.childCount === 0) { //Base case and malformed statements
+			return parsedObject;
+		}
+		// In a well formed expression, Argument and propertyValue tokens should have exactly one child, from their definitions in mongo.g4
+		// The only difference in types of children between PropertyValue and argument tokens is the functionCallContext that isn't handled at the moment.
+		let child: ParseTree = ctx.children[0];
+		if (child instanceof mongoParser.LiteralContext) {
```

Comment:
I'd create some util funtions:
  filterType<T>(array, callback) -> filter objects which are instanceof T
  findType<T>(array, callback) { return filterType<T>(array, callback)[0]; }

Reply:
I'm finding implementing it to be more code than it's worth. Let's sync up tomorrow about this.